### DISCUSSION TITLE
Livewire making multiple mount function querries on every subsequent request.

### DISCUSSION THREAD

**Discussion by scss-brad** [livewire/livewire#4757] [1. Help]
While logging MYSQL queries, I noticed the Livewire component is running the mount() function on each request. I can reproduce this issue with code below. Laravel 8.83.2, Livewire 2.10.4, with Breeze installed. Am I doing something wrong?

LiveWire Component:
```
<?php

namespace App\Http\Livewire;

use Livewire\Component;
use App\Models\TestModel;

class TestLivewire extends Component
{
    public TestModel $thing;
    protected $listeners = [''];
    protected $rules = [];

    
    public function mount($id)
    {
        $this->thing = TestModel::find($id);
    }

    public function render()
    {
        return view('testview');
    }

    public function button()
    {

    }
}
```

Model:
```
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;


class TestModel extends Model
{
    use HasFactory;

    protected $table = 'test';
    protected $primaryKey = "id";

}
```

Route:
```
Route::middleware(['auth'])->group(function () 
{
Route::get('/test/{id}', TestLivewire::class);
}
```
testview
```
<div>
<button class='btn btn-blue' wire:click='button'>Click</button>
</div>
```
app.blade
```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="csrf-token" content="{{ csrf_token() }}">
        <title>{{ config('app.name', 'Laravel') }}</title>
  
        <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Nunito:wght@400;600;700&display=swap">
        <link rel="stylesheet" href="{{ asset('css/app.css') }}">
        <script src="{{ asset('js/app.js') }}" defer></script>
        <livewire:styles />
    </head>
    <body class="font-sans bg-gray-200">
        <div class="bg-gray-100 p-2">
            @include('layouts.navigation')

            <!-- Page Heading -->
            <header class="bg-white shadow">
                <div class="py-6 px-4 sm:px-6 lg:px-8">
                    SMS
                </div>
            </header>

            <!-- Page Content -->
            <main class="">
                {{ $slot }}
            </main>
        </div>
        <livewire:scripts />
    </body>
</html>)
```
Here is the log view with time stamps. The first set is initial page load. I counted 5 seconds in my head (clearly I am slow) then clicked the button. The second set seems to show the mount function firing again, system wide.
```
[2022-03-03 04:25:17]
select * from `users` where `id` = ? limit 1 [3]

[2022-03-03 04:25:17]
select * from `test` where `test`.`id` = ? limit 1 [1]

[2022-03-03 04:25:25]
select * from `users` where `id` = ? limit 1 [3]

[2022-03-03 04:25:25]
select * from `test` where `test`.`id` = ? limit 1 [1]
```
Notice the data is not being used anywhere in the view and the click of the button leads to a function that performs nothing.

I just want to know if this is standard functionality to query the database for no reason during any request. I thought the mount() function should only run once. The same thing happens when applying wire:model hook and creating a input event.

The problem significantly snowballs out of control once you start eager loading multiple relationships using with() and/or using nested Livewire components in a foreach loop - each nested component using parent component public variables causes the query to repeat itself in the loop.

Any advice or thoughts would be helpful. 

EDIT::: I simplified app.blade, removed authentication in the route. Same results (minus the user model).

**Comment by silnarm** [comment]
`mount` is only running once, on subsequent requests public property hydration will result in similar (or the same) queries being run, to get the 'starting state' for any bound models.

**Reply by scss-brad** [reply]
Seems wildly inefficient.

For example, I thought you could use functions like push() to add models to collections without making another database query to return the new results - if hydrate() is making the query regardless you might as well command a refresh since there are no advantages to using push() to avoid another MYSQL query. I assume that if you directly command another query, it would result in two sets of queries being run.

**Comment by joshhanley** [comment]
@scss-brad it's less about efficiency, and more that, that is just how PHP works.

Due to PHP being a stateless environment, when a new Livewire request comes in, the models need to come from somewhere, so Livewire makes a database request to get them to "rehydrate" them.

If you don't want that happening, I'd suggest just using plain arrays and converting your models to arrays before storing them in public properties.

Then when you are ready to persist the changes, you have to parse the arrays and update/create models as required.

Hope this helps!
